test/packages_test.rb

```diff
@@ -836,4 +847,32 @@ def pattern(properties = {})
       end
     end
   end
+
+  describe "#Reset" do
```

Comment:
Just a nitpicking: I know that a test should be self-describing, but could you add more comments to this behavior? Those `it` parts are somehow `not enough` understandable to me. You could maybe define some `when` clauses.